bu_passport/scripts/cfa_event_scraper.py

```diff
@@ -0,0 +1,319 @@
+import re
+import requests
+from datetime import datetime
+import hashlib
+
+from dataclasses import dataclass, field
+from typing import List, Optional, Tuple
+
+import firebase_admin
+from firebase_admin import credentials, firestore
+
+from bs4 import BeautifulSoup
+import pytz
+from urllib.parse import urlparse, parse_qs
+
+
+@dataclass
+class CFAEvent:
+    event_id: str = ""
+    event_id_hex: str = ""
+    title: Optional[str] = ""
+    description: Optional[str] = ""
+    categories: List[str] = field(default_factory=list)
+    location: Optional[str] = ""
+    photo: Optional[str] = ""
+    points: int = 0  # Default points to 0
+    start_time: Optional[datetime] = None
+    end_time: Optional[datetime] = None
+    event_url: Optional[str] = ""
+    detail_url: Optional[str] = ""
+
+    def to_dict(self) -> dict:
+        return {
+            "eventID": self.event_id,
+            "eventTitle": self.title,
+            "eventCategories": self.categories,
+            "eventLocation": self.location,
+            "eventStartTime": self.start_time,
+            "eventEndTime": self.end_time,
+            "eventURL": self.event_url,
+            "eventDescription": self.description,
+            "eventPhoto": self.photo,
+            "eventPoints": 30,
+            "savedUsers": [],
+        }
+        
+    def to_dict_exist(self) -> dict:
```

Comment:
A little nitpick about general good practices: The `_exist` suffix here is related to how you're saving this event to the database, if it exists in the DB you use this method. That's a detail that this object shouldn't be concerned about. Instead think about what this object is doing and be more specific, i.e `to_dict` and `to_dict_with_saved_users` might more descriptive to external callers.

As a second note, these two methods are VERY overlapping, think about how you can reuse them so you don't have to repeat so much code? This is called the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle!